x/mongo/driver/connstring/connstring.go

```diff
@@ -295,6 +295,17 @@ func (p *parser) parse(original string) error {
 		}
 	}
 
+	// A positive "srvMaxHosts" value is not allowed with "loadbalanced=true" or
```

Comment:
I'd recommend duplicating this logic in `clientoptions.go` as we do for validation of the `LoadBalanced` option. This isn't ideal, but I think we'll have to duplicate this sort of logic until we get to the "stabilize connstring package" epic, which should hopefully clean up the interactions between `ConnString` and `Client`. Let's falso add some tests to `client_options_test.go` while we're there.